direct/nn/conjgradnet/conjgradnet_engine.py

```diff
@@ -0,0 +1,45 @@
+# coding=utf-8
+# Copyright (c) DIRECT Contributors
+
+from typing import Any, Callable, Dict, Optional, Tuple
+
+import torch
+from torch import nn
+
+from direct.config import BaseConfig
+from direct.nn.mri_models import MRIModelEngine
+
+
+class ConjGradNetEngine(MRIModelEngine):
+    def __init__(
+        self,
+        cfg: BaseConfig,
+        model: nn.Module,
+        device: str,
+        forward_operator: Optional[Callable] = None,
+        backward_operator: Optional[Callable] = None,
+        mixed_precision: bool = False,
+        **models: nn.Module,
+    ):
+        """Inits :class:`ConjGradNetEngine`."""
+        super().__init__(
+            cfg,
+            model,
+            device,
+            forward_operator=forward_operator,
+            backward_operator=backward_operator,
+            mixed_precision=mixed_precision,
+            **models,
+        )
+
+    def forward_function(self, data: Dict[str, Any]) -> Tuple[torch.Tensor, None]:
```

Comment:
The dict can maybe be replaced with either a namedtuple or a typeddict. The latter supports optional arguments (you can leave them out) in the latest python, but not sure it's already the moment for this. Anyway, it should be on the roadmap

Reply:
Same as operators.